src/Microsoft.Diagnostics.Tracing.Eventing/EventPipeClient.cs

```diff
@@ -0,0 +1,194 @@
+
+                    var sw = new BinaryWriter(namedPipe);
+                    sw.Write(serializedConfiguration);
+
+                    var br = new BinaryReader(namedPipe);
+                    return br.ReadUInt64();
+                }
+            }
+            else
+            {
+                throw new PlatformNotSupportedException("TODO: Get the ApplicationGroupId to form the string: 'dotnetcore-diagnostic-{processId}-{ApplicationGroupId}-socket'");
+            }
+        }
+
+        public static ulong DisableTracingToFile(int processId, ulong sessionId)
+        {
+            var header = new MessageHeader {
+                RequestType = DiagnosticMessageType.Disable,
+                Pid = (uint)Process.GetCurrentProcess().Id,
+            };
+
+            byte[] sessionIdInBytes;
+            using (var stream = new MemoryStream())
+            {
+                using (var sw = new BinaryWriter(stream))
+                {
+                    sw.Write((uint)header.RequestType);
+                    sw.Write(header.Pid);
+
+                    sw.Write(sessionId);
+                    sw.Flush();
+                    stream.Position = 0;
+                    sessionIdInBytes = new byte[stream.Length];
+                    stream.Read(sessionIdInBytes, 0, sessionIdInBytes.Length);
+                }
+                Console.WriteLine($"Serialized data is {sessionIdInBytes.Length} bytes vs {sizeof(ulong)}.");
+            }
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                var pipeName = $"dotnetcore-diagnostic-{processId}";
+                using (var namedPipe = new NamedPipeClientStream(
+                    ".", pipeName, PipeDirection.InOut, PipeOptions.None, TokenImpersonationLevel.Impersonation))
+                {
+                    namedPipe.Connect((int)TimeSpan.FromSeconds(20).TotalMilliseconds);
+
+                    var sw = new BinaryWriter(namedPipe);
+                    sw.Write(sessionIdInBytes);
+                    return sessionId;
+                }
+            }
+            else
+            {
+                using (var socket = new Socket(AddressFamily.Unix, SocketType.Stream, ProtocolType.Unspecified))
+                {
+                    var remoteEP = new UnixDomainSocketEndPoint("/tmp/dotnetcore-diagnostic-1234.socket");
+                    socket.Bind(remoteEP);
+                    socket.Connect(remoteEP);
+
+                    socket.Send(sessionIdInBytes);
+                    return sessionId;
+                }
+            }
+        }
+
+        private static int GetSizeInBytes(string str)
+        {
+            var strLength = str == null ? 0 : Encoding.Unicode.GetByteCount(str + '\0');
+            return Marshal.SizeOf(typeof(int)) + strLength;
+        }
+
+        private static int GetByteCount(ProviderConfiguration providerConfiguration)
```

Comment:
Wouldn't it be nicer to make this a method under `ProviderConfiguration`?

Reply:
Not used :)